debezium-embedded/src/main/java/io/debezium/embedded/ConvertingEngineBuilder.java

```diff
@@ -197,13 +204,23 @@ private Converter createConverter(Class<? extends SerializationFormat<?>> format
         converterConfig = commonConverterConfig.edit().with(converterConfig).build();
 
         if (isFormat(format, Json.class)) {
-            converterConfig = converterConfig.edit().withDefault(FIELD_CLASS, "org.apache.kafka.connect.json.JsonConverter").build();
+            if (converterConfig.hasKey(APICURIO_SCHEMA_REGISTRY_URL_CONFIG)) {
+                converterConfig = converterConfig.edit().withDefault(FIELD_CLASS, "io.apicurio.registry.utils.converter.ExtJsonConverter").build();
+            }
+            else {
+                converterConfig = converterConfig.edit().withDefault(FIELD_CLASS, "org.apache.kafka.connect.json.JsonConverter").build();
+            }
         }
         else if (isFormat(format, CloudEvents.class)) {
             converterConfig = converterConfig.edit().withDefault(FIELD_CLASS, "io.debezium.converters.CloudEventsConverter").build();
         }
         else if (isFormat(format, Avro.class)) {
-            converterConfig = converterConfig.edit().withDefault(FIELD_CLASS, "io.confluent.connect.avro.AvroConverter").build();
+            if (converterConfig.hasKey(APICURIO_SCHEMA_REGISTRY_URL_CONFIG)) {
+                converterConfig = converterConfig.edit().withDefault(FIELD_CLASS, "io.apicurio.registry.utils.converter.AvroConverter").build();
+            }
+            else {
+                converterConfig = converterConfig.edit().withDefault(FIELD_CLASS, "io.confluent.connect.avro.AvroConverter").build();
+            }
         }
         else if (isFormat(format, Protobuf.class)) {
```

Comment:
Shouldn't we use it for Protobuf too?

Reply:
Apicurio doesn't provide any protobuf converter so the default is confluent's `ProtobufConverter`